### PR TITLE
Add new draft of a DOSP pattern

### PR DESCRIPTION
**Pull request by dosumis**
This commit intends to add a draft for a new DOSP pattern.
If applied, this commit is related to #53.

### CODE REVIEW

**Comment by dosumis** [comment]
Hmmm - not seeing pattern tests running.   @matentzn - is this correct?  I hacked the makefile slightly on this PR as a temp fix to get tests running early https://github.com/obophenotype/uberon/pull/2231/files#diff-1376bbad1ef43bed6c9aed84b7d52471e63c87277b2b3f72ed0a5bdef481b638L64.  Does ODK need some fix?

**Comment by matentzn** [comment]
No its an outdated version of ODK Makefile. Newer ODKs run the checks _of the yaml files_ every time. For OBA, it makes sense to run the entire dosdp pipeline though, so I have activated that now.

**Comment by matentzn** [comment]
https://github.com/obophenotype/bio-attribute-ontology/commit/d02562e9cc7d4b1a1550e0626f91974d40b333dc

**Comment by dosumis** [comment]
Should this come through ODK update when DOSDP = true ?

**Comment by matentzn** [comment]
I have taken the stance that we should minimise our load on GitHub actions as much as possible. The validation of the yaml files is run every time - the pattern generation is not, with the same logic as the not re-running imports. But yeah, its a very thin line here.. I can be convinced either way.

**Comment by dosumis** [comment]
> I have taken the stance that we should minimise our load on GitHub actions as much as possible. The validation of the yaml files is run every time - the pattern generation is not, with the same logic as the not re-running imports. But yeah, its a very thin line here.. I can be convinced either way.

Maybe bare minimum should be python pattern checks.  Very little load on GA from that.

**Comment by rays22** [comment]
> Relations (OPs) should be single quoted.

@dosumis Does the yaml look OK? The QC has passed, but I wanted to make sure that the new pattern also makes sense.

**Comment by dosumis** [comment]
> @dosumis Does the yaml look OK? The QC has passed, but I wanted to make sure that the new pattern also makes sense.

Looks good to me. Can you also make a start on the TSV in this PR?  Doesn't need to be complete, but would be good to have a few test rows so we can review the results of a full build.

**Comment by dosumis** [comment]
@matentzn  - still don't see pattern checks running here: https://github.com/obophenotype/bio-attribute-ontology/runs/5065418876?check_suite_focus=true

**Comment by matentzn** [comment]
Checks are in place now, update from master.

**Comment by dosumis** [comment]
Please can we make it a priority to unblock this ASAP.  We need to try to wrap up the basic work on this by the next GWAS call (2nd March).

@rays22  please update branch from master.  This should => you Python checks on patterns on this PR.  Let me know if anything in the reports is unclear or you can't manage to fix. If there are still problems with the CI check -  it's easy to to run the Python checks locally. You can install from PyPi https://pypi.org/project/dosdp/

**Comment by rays22** [comment]
I had to remove the following 5 lines from the tsv filler, because they all caused `ERROR	multiple_labels	urn:dosdp:defined_class	rdfs:label`.
I am wondering what the underlying problem is. @dosumis , do you have any suggestions?


defined_class | defined_class_name | chemical | chemical_name | <http://www.geneontology.org/formats/oboInOwl#created_by>
-- | -- | -- | -- | --
2040030 |   | CHEBI:30879 | alcohol | <https://orcid.org/0000-0002-7073-9172>\|<https://orcid.org/0000-0001-8314-2140>
2040036 |   | CHEBI:33281 | antimicrobial agent | <https://orcid.org/0000-0002-7073-9172>\|<https://orcid.org/0000-0001-8314-2140>
2040047 |   | CHEBI:35523 | bronchodilator agent | <https://orcid.org/0000-0002-7073-9172>\|<https://orcid.org/0000-0001-8314-2140>
2040088 |   | CHEBI:50858 | corticosteroid | <https://orcid.org/0000-0002-7073-9172>\|<https://orcid.org/0000-0001-8314-2140>
2040016 |   | CHEBI:23888 | drug | <https://orcid.org/0000-0002-7073-9172>\|<https://orcid.org/0000-0001-8314-2140>

I am still getting similar errors:

```
First 5 violations:
ERROR	multiple_labels	urn:dosdp:defined_class	rdfs:label	trait in response to drug
ERROR	multiple_labels	urn:dosdp:defined_class	rdfs:label	trait in response to fatty acid
ERROR	multiple_labels	urn:dosdp:defined_class	rdfs:label	trait in response to glucocorticoid
ERROR	multiple_labels	urn:dosdp:defined_class	rdfs:label	trait in response to growth hormone
ERROR	multiple_labels	urn:dosdp:defined_class	rdfs:label	trait in response to http://purl.obolibrary.org/obo/CHEBI_10033
```

**Comment by matentzn** [comment]
Did you check these classes do not already exist in one of the other tables?

**Comment by rays22** [comment]
> Did you check these classes do not already exist in one of the other tables?

No, they do not exist in any of the other local tables in the repo.

**Comment by rays22** [comment]
> 

@matentzn: the issue was the missing `OBA:` curies in front of the new class IDs. Could you take another look if it looks OK now?

File: src/patterns/dosdp-patterns/response_to_chemical_stimulus_trait.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+pattern_name: response_to_chemical_stimulus_trait
+
+pattern_iri: http://purl.obolibrary.org/obo/oba/patterns/response_to_chemical_stimulus_trait.yaml
+
+description: 'A trait that affects the biological response to a chemical.'
```
Comment (matentzn) [inline src/patterns/dosdp-patterns/response_to_chemical_stimulus_trait.yaml:6]: `affects` does not seem like the right word here, but ok for now.. Just keep in mind to update the language moving forward.

**Review by matentzn — APPROVED** [review]

File: src/patterns/dosdp-patterns/response_to_chemical_stimulus_trait.yaml (hunk 1 above)
Comment (rays22) [reply src/patterns/dosdp-patterns/response_to_chemical_stimulus_trait.yaml:6]: @matentzn How about
`A trait of a biological response to a chemical.`
? Would that wording be better?

Comment (matentzn) [reply src/patterns/dosdp-patterns/response_to_chemical_stimulus_trait.yaml:6]: Again, I dont know whether "trait of a response" is the right grammar.. Just leave it for now, don't worry. Ask David in your next meeting with him

Comment (dosumis) [reply src/patterns/dosdp-patterns/response_to_chemical_stimulus_trait.yaml:6]: I think 'affects' is good enough.  I can't think of a non-confusing alternative.  I think this comes down to the basic problem/confusion of 'phenotypic effect' vs 'abstract trait'/measurement.  The association with a variant in the GWAS catalog is certainly recording some effect on response to stimulus with the chemical in question.

File: src/patterns/dosdp-patterns/response_to_chemical_stimulus_trait.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - chemical
+
+annotations: 
+  - annotationProperty: exact_synonym
+    text: "response to %s trait"
+    vars:
+      - chemical
+      
+def:
+  text: "A trait that affects to the response to a stimulus with %s."
```
Comment (dosumis) [inline src/patterns/dosdp-patterns/response_to_chemical_stimulus_trait.yaml:40]: remove 'to' from 'affects to'

**Pull request merged**